src/cmd/moby/run_qemu.go

```diff
@@ -53,78 +75,185 @@ func runQemu(args []string) {
 		log.Warnf("Both -iso and -uefi have been used")
 	}
 
-	// Before building qemu arguments, check qemu is in the $PATH
-	qemuBinPath := "qemu-system-" + *qemuArch
+	config := QemuConfig{
+		Prefix:   prefix,
+		ISO:      *qemuIso,
+		UEFI:     *qemuUEFI,
+		Kernel:   *qemuKernel,
+		GUI:      *qemuGUI,
+		DiskPath: *qemuDiskPath,
+		DiskSize: *qemuDiskSize,
+		FWPath:   *qemuFWPath,
+		Arch:     *qemuArch,
+		CPUs:     *qemuCPUs,
+		Memory:   *qemuMem,
+	}
+
+	config, qemuArgs := buildQemuCmdline(config)
+
+	var err error
+	if config.Containerized {
+		err = runQemuContainer(config, qemuArgs)
+	} else {
+		err = runQemuLocal(config, qemuArgs)
+	}
+	if err != nil {
+		log.Fatal(err.Error())
+	}
+}
+
+func runQemuLocal(config QemuConfig, args []string) error {
+	if config.DiskPath != "" {
+		// If disk doesn't exist then create one
+		if _, err := os.Stat(config.DiskPath); os.IsNotExist(err) {
```

Comment:
I noticed this pattern on @thebsdbox's PR too - I'm inclined to bail on this error if it is not `nil` and is also `os.IsNotExist(err) == false` for extra safety and clearer debugging.

Also it would make it consistent with how you handle the OVMF firmware below: https://github.com/linuxkit/linuxkit/pull/1672/files#diff-7ea76eb7a56000c30ad47347644f6c66R123